roofit/roofitcore/src/RooFit/CodegenContext.cxx

```diff
@@ -104,6 +104,15 @@ void CodegenContext::addVecObs(const char *key, int idx)
       _vecObsIndices[namePtr] = idx;
 }
 
+int CodegenContext::observableIndexOf(RooAbsArg const *arg) const
```

Comment:
```suggestion
int CodegenContext::observableIndexOf(RooAbsArg const &arg) const
```
That's a pretty useful function to add! Could you change the signature of the function such that it takes `arg` by const reference (e.g. `const &`), to be more consistent with the other functions in this class?

Unlike references, pointers can be `nullptr`, which would not make any sense in this function because the pointer is always dereferenced n `arg->namePtr()`. Using references makes it explicit to the user and compiler that the argument can't be "null", adding safety to your code.

More on this also in chapter 12 of the `learncpp.com` guide (another recommended read :slightly_smiling_face: ):
https://www.learncpp.com/cpp-tutorial/introduction-to-compound-data-types/